### ISSUE TITLE
Some Docker Registries impose page size limit, add configurable pageSize to ImageRepository 

### ISSUE THREAD

**Issue by carl-reverb** [fluxcd/image-reflector-controller#585]
Hello, so we have run into an issue that makes us unable to use Image Automation with some Docker Registries which limit PageSize on the `v2/.../.../tags/list` endpoint.

```
scan failed: GET https://quay.io/v2/pganalyze/collector/tags/list?n=1000: TOO_MANY_TAGS_REQUESTED: Too many tags requested; requested limit 1000 is too high, max is 100
```

In order to resume using Image Automation with Quay.io, we'd need some way of setting `pageSize` on the `ImageRepository`.

The underlying `go-containerregistry.remote` already has a `withPageSize` function. 

https://github.com/google/go-containerregistry/blob/main/pkg/v1/remote/options.go#L292

**Comment by gionn** [comment]
just tried reaching Quay on X https://x.com/gionn/status/1804162576382664972

**Comment by stefanprodan** [comment]
Major breaking change shipped on a Friday 😱  This also breaks source-controller for HelmCharts and OCIRepositories with semver ranges.

**Comment by YoninL** [comment]
> This also breaks source-controller for HelmCharts and OCIRepositories with semver ranges.

This is also quay.io images limited, right? Not others.

**Comment by makkes** [comment]
> > This also breaks source-controller for HelmCharts and OCIRepositories with semver ranges.
> 
> This is also quay.io images limited, right? Not others.

yes

**Comment by stefanprodan** [comment]
> In order to resume using Image Automation with Quay.io, we'd need some way of setting pageSize on the ImageRepository.

This shouldn't be needed. Quay should return a page with 100 tags (their max page size) along with a link header for the next page. This is how all the other registries behave.

**Comment by joedborg** [comment]
Is there any way (except recompiling IRC) to get the maximum to 100 in the interim?

**Comment by makkes** [comment]
[We're not the only ones downstream that are affected by this change](https://github.com/google/go-containerregistry/issues/1963).

**Comment by makkes** [comment]
> Is there any way (except recompiling IRC) to get the maximum to 100 in the interim?

No, the limit parameter is hard-coded to 1000 and current IRC has no way to change that.

**Comment by syed** [comment]
We (quay.io) have deployed a fix where you'll get the max tags instead of a 417. Sorry for the Friday disruption 😅

**Comment by stefanprodan** [comment]
Thanks @syed 

Due to exponential backoff in Flux, the ImageRepository reconcilers will take a long time to recover. For a faster recovery, restar image-reflector-controller by deleting its pod.
